lib/entry-index.js

```diff
@@ -85,7 +103,13 @@ async function compact (cache, key, matchFn, opts = {}) {
   // write the file atomically
   await disposer(setup(), teardown, write)
 
-  return newEntries.map((entry) => formatEntry(cache, entry, true))
+  // we reverse the list we generated such that the newest
+  // entries come first in order to make looping through them easier
+  // the true passed to formatEntry tells it to keep null
+  // integrity values, if they made it this far it's because
+  // the user specified a metadata field in keepMissingIntegrityWhenSet
```

Comment:
Is `keepMissingIntegrityWhenSet` still a thing?  I thought the caller just provides their own `validateEntry` function to do that.